{"uuid": "3f3f02c1-24cd-4716-b162-94826d84ce09", "children": ["4ffb0583-b18d-4b03-904e-2af231212417"], "befores": [{"name": "test_customer_data", "status": "broken", "statusDetails": {"message": "NameError: name 'random' is not defined. Did you forget to import 'random'?\n", "trace": "  File \"/Users/aleks/QA-hw/opencart_db_framework/.venv/lib/python3.13/site-packages/pluggy/_callers.py\", line 167, in _multicall\n    teardown.throw(outcome._exception)\n    ~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^\n  File \"/Users/aleks/QA-hw/opencart_db_framework/.venv/lib/python3.13/site-packages/_pytest/setuponly.py\", line 36, in pytest_fixture_setup\n    return (yield)\n            ^^^^^\n  File \"/Users/aleks/QA-hw/opencart_db_framework/.venv/lib/python3.13/site-packages/pluggy/_callers.py\", line 103, in _multicall\n    res = hook_impl.function(*args)\n  File \"/Users/aleks/QA-hw/opencart_db_framework/.venv/lib/python3.13/site-packages/_pytest/fixtures.py\", line 1140, in pytest_fixture_setup\n    result = call_fixture_func(fixturefunc, request, kwargs)\n  File \"/Users/aleks/QA-hw/opencart_db_framework/.venv/lib/python3.13/site-packages/_pytest/fixtures.py\", line 898, in call_fixture_func\n    fixture_result = fixturefunc(**kwargs)\n  File \"/Users/aleks/QA-hw/opencart_db_framework/tests/test_customer.py\", line 23, in test_customer_data\n    \"firstname\": generate_random_string(),\n                 ~~~~~~~~~~~~~~~~~~~~~~^^\n  File \"/Users/aleks/QA-hw/opencart_db_framework/tests/test_customer.py\", line 12, in generate_random_string\n    return ''.join(random.choices(string.ascii_letters, k=length))\n                   ^^^^^^\n"}, "start": 1746280502589, "stop": 1746280502589}], "start": 1746280502589, "stop": 1746280502675}